server/validation.ts

```diff
@@ -165,3 +167,19 @@ export const assertCollectionPermission = (
 ) => {
   assertIn(value, [...Object.values(CollectionPermission), null], message);
 };
+
+/**
+ * Checks if documentId is valid
+ *
+ * @param documentId
+ * @returns true if documentId is valid else false
+ */
+export const isValidDocumentId = (documentId: string) =>
+  isUUID(documentId) || SLUG_URL_REGEX.test(documentId);
+
+export const INDEX_REGEX = new RegExp("^[\x20-\x7E]+$");
+
+// Validation messages
+export const INVALID_DOCUMENT_ID = "Must be uuid or url slug";
+
+export const INVALID_INDEX = "Must be between x20 to x7E ASCII";
```

Comment:
Not a huge fan of these capitalized constants, I think they look ugly and the validation and message not clearly linked. How about some tiny classes...

```
export class ValidateIndex {
  public static refine = new RegExp("^[\x20-\x7E]+$");
  public static message = "Must be between x20 to x7E ASCII";
}
```